src/System.Management.Automation/engine/SessionStateDriveAPIs.cs

```diff
@@ -894,21 +892,19 @@ private bool IsAStaleVhdMountedDrive(PSDriveInfo drive)
             //  3. Unmount the VHD in session 'A'.
             // The drive pointing to VHD in session 'B' gets detected as DriveType.NoRootDirectory
             // after the VHD is removed in session 'A'.
-            if (drive != null && !string.IsNullOrEmpty(drive.Name) && drive.Name.Length == 1)
+            if (drive is { Name.Length: 1 })
```

Comment:
I'd prefer to avoid using patterns since we haven't nullability check enabled.
I guess something classic like `drive?.Name?.Length == 1` would work here.